src/tools/schema2kotlin.ts

```diff
@@ -75,19 +75,17 @@ ${this.opts.wasm ? 'import arcs.sdk.wasm.*' : 'import arcs.core.storage.api.toPr
       if (this.opts.wasm) {
         handleInterfaceType = `Wasm${handleConcreteType}Impl<${entityType}>`;
       } else {
-        switch (connection.direction) {
-          case 'reads writes':
-            handleInterfaceType = `ReadWrite${handleConcreteType}<${entityType}>`;
-            break;
-          case 'writes':
-            handleInterfaceType = `Writable${handleConcreteType}<${entityType}>`;
-            break;
-          case 'reads':
-            handleInterfaceType = `Readable${handleConcreteType}<${entityType}>`;
-            break;
-          default:
+        if (connection.direction !== 'reads' && connection.direction !== 'writes' && connection.direction !== 'reads writes') {
```

Comment:
Do we allow "writes reads"? I know we always say "reads writes", but what should happen if a developer gets them backwards?

Reply:
Interesting idea, currently we do not (and my guess is that it would introduce confusion) but we could and I believe the type checking would support it well.

Reply:
Actually I don't think I answered your question properly. Currently the manifest syntax doesn't understand writes reads. We could add support for this (reordering keywords) with fairly little pain (just in the parser). If you think it worthwhile I'll write up a PR and run it past a few people.